### PR TITLE
support sudo switch and other improvements.

### PR DESCRIPTION
**Pull request by squirrelsc**
see commit messages for details.

### CODE REVIEW

File: lisa/node.py [hunk 1]
```diff
@@ -161,6 +175,25 @@ def is_linux(self) -> bool:
         self.initialize()
         return self.os.is_linux
 
+    @property
+    def support_sudo(self) -> bool:
+        self.initialize()
+
+        # check if sudo supported
+        if self.is_linux and self._support_sudo is None:
+            process = self._execute("command -v sudo", shell=True, no_info_log=True)
```
Comment (LiliDeng) [inline lisa/node.py:184]: can we use like sudo -V, then check whether it exist or not?

File: lisa/node.py [hunk 2]
```diff
@@ -161,6 +175,25 @@ def is_linux(self) -> bool:
...
+            result = process.wait_result(10)
```
Comment (LiliDeng) [inline lisa/node.py:185]: does it return immediately?

Comment (squirrelsc) [reply lisa/node.py:185]: yes, it's synchronous call, return immediately. If it spends more than 10 seconds, there must be some problem.

File: lisa/node.py (hunk 1 above)
Comment (squirrelsc) [reply lisa/node.py:184]: what can `sudo -V` help? Is there any compatibility issue we found on different sudo version?

Comment (LiliDeng) [reply lisa/node.py:184]: in case, command 'command' doesn't exist, sudo -V is to display the version.

Comment (squirrelsc) [reply lisa/node.py:184]: `command` is heavily used to detect every "command". If it's not here, no case can pass. So it doesn't need to use a different way to `sudo` only.

**Review by LiliDeng — APPROVED** [review]

**Pull request merged**